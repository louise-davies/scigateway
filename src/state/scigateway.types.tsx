import { Step } from 'react-joyride';
import { Theme } from '@material-ui/core/styles/createMuiTheme';

export const AuthSuccessType = 'scigateway:auth_success';
export const AuthFailureType = 'scigateway:auth_failure';
export const LoadingAuthType = 'scigateway:loading_auth';
export const LoadedAuthType = 'scigateway:loaded_auth';
export const LoadAuthProviderType = 'scigateway:loading_auth_provider';
export const LoadScheduledMaintenanceStateType =
  'scigateway:load_scheduled_maintenance_state';
export const LoadMaintenanceStateType = 'scigateway:load_maintenance_state';
export const InvalidateTokenType = 'scigateway:api:invalidate_token';
export const ConfigureStringsType = 'scigateway:configure_strings';
export const LoginType = 'scigateway:login';
export const NotificationType = 'scigateway:api:notification';
export const RegisterRouteType = 'scigateway:api:register_route';
export const RequestPluginRerenderType = 'scigateway:api:plugin_rerender';
export const SendThemeOptionsType = 'scigateway:api:send_themeoptions';
export const LoadDarkModePreferenceType =
  'scigateway:load_dark_mode_preference';
export const SignOutType = 'scigateway:signout';
export const ToggleDrawerType = 'scigateway:toggledrawer';
export const DismissNotificationType = 'scigateway:dismissnotification';
export const ConfigureFeatureSwitchesType = 'scigateway:feature_switches';
export const AddNotificationsType = 'scigateway:add_notification';
export const SiteLoadingType = 'scigateway:site_loading';
export const ConfigureAnalyticsType = 'scigateway:configure_analytics';
export const InitialiseAnalyticsType = 'scigateway:initialise_analytics';
export const ToggleHelpType = 'scigateway:toggle_help';
export const AddHelpTourStepsType = 'scigateway:add_help_tour_steps';
export const RegisterStartUrlType = 'scigateway:register_start_url';

export interface NotificationPayload {
  message: string;
  severity: string;
}

export interface AddNotificationsPayload {
  notifications: NotificationPayload[];
}

export interface AppStrings {
  [id: string]: string;
}

export interface ConfigureStringsPayload {
  res: ApplicationStrings;
}

export interface ApplicationStrings {
  [section: string]: AppStrings;
}

export interface FeatureSwitchesPayload {
  switches: FeatureSwitches;
}

export interface FeatureSwitches {
  showContactButton: boolean;
  showHelpPageButton: boolean;
}

export interface StartUrlPayload {
  startUrl: string;
}

export interface RegisterRoutePayload {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface PluginConfig {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface GroupedPlugins {
  [section: string]: PluginConfig[];
}

export interface LoginPayload {
  username: string;
  password: string;
}

export interface AccessPayload {
  authorisation: boolean;
  accessToken: string;
  error: null;
}

export interface AuthorisedPayload {
  token: string;
}

export interface SendThemeOptionsPayload {
  theme: Theme;
  broadcast: boolean;
}

export interface LoadDarkModePreferencePayload {
  darkMode: boolean;
}

export interface DismissNotificationPayload {
  index: number;
}

export interface AuthProviderPayload {
  authProvider: string;
  authUrl: string | undefined;
}

export interface ScheduledMaintenanceStatePayLoad {
  scheduledMaintenance: ScheduledMaintenanceState;
}

export interface ScheduledMaintenanceState {
  show: boolean;
  message: string;
}

export interface MaintenanceStatePayLoad {
  maintenance: MaintenanceState;
}

export interface MaintenanceState {
  show: boolean;
  message: string;
}

export interface SiteLoadingPayload {
  loading: boolean;
}

export interface ConfigureAnalyticsPayload {
  id: string;
}

export interface AddHelpTourStepsPayload {
  steps: Step[];
}
