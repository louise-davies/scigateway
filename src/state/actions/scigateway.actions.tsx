import axios from 'axios';
import { Action, AnyAction } from 'redux';
import log from 'loglevel';
import {
  ConfigureStringsType,
  ConfigureStringsPayload,
  ToggleDrawerType,
  AuthFailureType,
  AuthSuccessType,
  ApplicationStrings,
  SignOutType,
  FeatureSwitchesPayload,
  ConfigureFeatureSwitchesType,
  FeatureSwitches,
  LoadingAuthType,
  LoadedAuthType,
  DismissNotificationType,
  DismissNotificationPayload,
  RequestPluginRerenderType,
  AuthProviderPayload,
  LoadAuthProviderType,
  SiteLoadingType,
  SiteLoadingPayload,
  ConfigureAnalyticsPayload,
  ConfigureAnalyticsType,
  InitialiseAnalyticsType,
  ToggleHelpType,
  AddHelpTourStepsType,
  AddHelpTourStepsPayload,
  InvalidateTokenType,
  SendThemeOptionsType,
  SendThemeOptionsPayload,
  LoadDarkModePreferenceType,
  LoadDarkModePreferencePayload,
  StartUrlPayload,
  RegisterStartUrlType,
  ScheduledMaintenanceState,
  LoadScheduledMaintenanceStateType,
  ScheduledMaintenanceStatePayLoad,
  MaintenanceStatePayLoad,
  LoadMaintenanceStateType,
  MaintenanceState,
  NotificationType,
} from '../scigateway.types';
import { ActionType, ThunkResult, StateType } from '../state.types';
import loadMicroFrontends from './loadMicroFrontends';
import { push } from 'connected-react-router';
import { ThunkAction } from 'redux-thunk';
import { Step } from 'react-joyride';
import { Theme } from '@material-ui/core/styles/createMuiTheme';

export const configureStrings = (
  appStrings: ApplicationStrings
): ActionType<ConfigureStringsPayload> => ({
  type: ConfigureStringsType,
  payload: {
    res: appStrings,
  },
});

export const loadStrings = (path: string): ThunkResult<Promise<void>> => {
  return async (dispatch) => {
    await axios
      .get(path)
      .then((res) => {
        dispatch(configureStrings(res.data));
      })
      .catch((error) =>
        log.error(`Failed to read strings from ${path}: ${error}`)
      );
  };
};

export const loadFeatureSwitches = (
  featureSwitches: FeatureSwitches
): ActionType<FeatureSwitchesPayload> => ({
  type: ConfigureFeatureSwitchesType,
  payload: {
    switches: featureSwitches,
  },
});

export const registerStartUrl = (
  startUrl: string
): ActionType<StartUrlPayload> => ({
  type: RegisterStartUrlType,
  payload: {
    startUrl: startUrl,
  },
});

export const addHelpTourSteps = (
  steps: Step[]
): ActionType<AddHelpTourStepsPayload> => ({
  type: AddHelpTourStepsType,
  payload: {
    steps,
  },
});

export const loadAuthProvider = (
  authProvider: string,
  authUrl?: string
): ActionType<AuthProviderPayload> => ({
  type: LoadAuthProviderType,
  payload: {
    authProvider,
    authUrl,
  },
});

export const loadScheduledMaintenanceState = (
  scheduledMaintenance: ScheduledMaintenanceState
): ActionType<ScheduledMaintenanceStatePayLoad> => ({
  type: LoadScheduledMaintenanceStateType,
  payload: {
    scheduledMaintenance: scheduledMaintenance,
  },
});

export const loadMaintenanceState = (
  maintenance: MaintenanceState
): ActionType<MaintenanceStatePayLoad> => ({
  type: LoadMaintenanceStateType,
  payload: {
    maintenance: maintenance,
  },
});

export const loadingAuthentication = (): Action => ({
  type: LoadingAuthType,
});

export const loadedAuthentication = (): Action => ({
  type: LoadedAuthType,
});

export const unauthorised = (): Action => ({
  type: AuthFailureType,
});

export const authorised = (): Action => ({
  type: AuthSuccessType,
});

export const invalidToken = (): Action => ({
  type: InvalidateTokenType,
});

export const siteLoadingUpdate = (
  loading: boolean
): ActionType<SiteLoadingPayload> => ({
  type: SiteLoadingType,
  payload: {
    loading,
  },
});

export const configureAnalytics = (
  id: string
): ActionType<ConfigureAnalyticsPayload> => ({
  type: ConfigureAnalyticsType,
  payload: {
    id,
  },
});

export const initialiseAnalytics = (): Action => ({
  type: InitialiseAnalyticsType,
});

export const configureSite = (): ThunkResult<Promise<void>> => {
  return async (dispatch, getState) => {
    await axios
      .get(`/settings.json`)
      .then((res) => {
        const settings = res.data;

        // invalid settings.json - Use JSON.parse to give detailed error info
        if (typeof settings !== 'object') {
          throw Error('Invalid format');
        }

        if (settings['ga-tracking-id']) {
          dispatch(configureAnalytics(settings['ga-tracking-id']));
        }

        dispatch(
          loadAuthProvider(settings['auth-provider'], settings['authUrl'])
        );

        const loadingPromises = [];

        const provider = getState().scigateway.authorisation.provider;

        if (provider.fetchMaintenanceState) {
          provider.fetchMaintenanceState().then((maintenanceState) => {
            dispatch(loadMaintenanceState(maintenanceState));
          });
        }

        // after auth provider is set then the token needs to be verified
        // also attempt to auto login if the auth provider allows it
        if (provider.isLoggedIn()) {
          const verifyingLogin = provider
            .verifyLogIn()
            .then(() => {
              dispatch(authorised());
            })
            .catch(() => {
              if (provider.autoLogin) {
                dispatch(loadingAuthentication());
                loadingPromises.push(
                  provider
                    .autoLogin()
                    .then(() => {
                      dispatch(authorised());
                    })
                    .catch(() => {
                      dispatch(invalidToken());
                    })
                );
              } else {
                dispatch(invalidToken());
              }
            });

          loadingPromises.push(verifyingLogin);
        } else if (provider.autoLogin) {
          dispatch(loadingAuthentication());
          loadingPromises.push(
            provider
              .autoLogin()
              .then(() => {
                dispatch(authorised());
              })
              .catch(() => {
                dispatch(loadedAuthentication());
              })
          );
        }

        if (settings['features']) {
          dispatch(loadFeatureSwitches(settings['features']));
        }

        dispatch(addHelpTourSteps(settings['help-tour-steps']));

        if (settings['startUrl']) {
          dispatch(registerStartUrl(settings['startUrl']));
        }

        const uiStringResourcesPath = !settings['ui-strings'].startsWith('/')
          ? '/' + settings['ui-strings']
          : settings['ui-strings'];
        const loadingResources = dispatch(loadStrings(uiStringResourcesPath));
        loadingPromises.push(loadingResources);

        const loadingPlugins = loadMicroFrontends.init(settings.plugins);
        loadingPromises.push(loadingPlugins);

        return Promise.all(loadingPromises).then(() => {
          dispatch(siteLoadingUpdate(false));
        });
      })
      .catch((error) => {
        log.error(`Error loading settings.json: ${error.message}`);
      });

    // load dark mode preference from local storage into store
    // otherwise, fetch system preference
    const darkModeLocalStorage = localStorage.getItem('darkMode');
    if (darkModeLocalStorage) {
      const preference = darkModeLocalStorage === 'true' ? true : false;
      dispatch(loadDarkModePreference(preference));
    } else {
      const mq = window.matchMedia('(prefers-color-scheme: dark)');
      if (mq) dispatch(loadDarkModePreference(mq.matches));
    }

    const provider = getState().scigateway.authorisation.provider;
    if (provider.fetchScheduledMaintenanceState) {
      provider
        .fetchScheduledMaintenanceState()
        .then((scheduledMaintenanceState) => {
          dispatch(loadScheduledMaintenanceState(scheduledMaintenanceState));

          // Checking the state in the GET response because it does not get
          // loaded into the store before this check is performed
          if (scheduledMaintenanceState['show']) {
            displayScheduledMaintenanceBanner(
              scheduledMaintenanceState['message']
            );
          }
        });
    }
  };
};

const displayScheduledMaintenanceBanner = (message: string): void => {
  document.dispatchEvent(
    new CustomEvent('scigateway', {
      detail: {
        type: NotificationType,
        payload: {
          severity: 'warning',
          message: message,
        },
      },
    })
  );
};

export const toggleDrawer = (): Action => ({
  type: ToggleDrawerType,
});

export const toggleHelp = (): Action => ({
  type: ToggleHelpType,
});

export const signOut = (): ThunkAction<void, StateType, null, AnyAction> => (
  dispatch
) => {
  dispatch({ type: SignOutType });
  dispatch(push('/'));
};

export const verifyUsernameAndPassword = (
  username: string,
  password: string
): ThunkResult<Promise<void>> => {
  return async (dispatch, getState) => {
    // will be replaced with call to login API for authentification
    dispatch(loadingAuthentication());
    const authProvider = getState().scigateway.authorisation.provider;
    await authProvider
      .logIn(username, password)
      .then(() => {
        dispatch(authorised());

        // redirect the user to the original page they were trying to get to
        // the referrer is added by the redirect in routing.component.tsx
        const previousRouteState = getState().router.location.state;
        dispatch(
          push(
            previousRouteState && previousRouteState.referrer
              ? previousRouteState.referrer
              : '/'
          )
        );
      })
      .catch(() => {
        // probably want to do something smarter with
        // err.response.status (e.g. 403 or 500)
        dispatch(unauthorised());
      });
  };
};

export const requestPluginRerender = (): ActionType<{
  broadcast: boolean;
}> => ({
  type: RequestPluginRerenderType,
  payload: {
    broadcast: true,
  },
});

export const sendThemeOptions = (
  theme: Theme
): ActionType<SendThemeOptionsPayload> => ({
  type: SendThemeOptionsType,
  payload: {
    theme,
    broadcast: true,
  },
});

export const loadDarkModePreference = (
  darkMode: boolean
): ActionType<LoadDarkModePreferencePayload> => ({
  type: LoadDarkModePreferenceType,
  payload: {
    darkMode: darkMode,
  },
});

export const dismissMenuItem = (
  index: number
): ActionType<DismissNotificationPayload> => {
  return {
    type: DismissNotificationType,
    payload: {
      index,
    },
  };
};
