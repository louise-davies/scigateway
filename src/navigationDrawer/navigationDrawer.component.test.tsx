import React from 'react';
import configureStore from 'redux-mock-store';
import { RouterState } from 'connected-react-router';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import NavigationDrawer from './navigationDrawer.component';
import { initialState } from '../state/reducers/scigateway.reducer';
import { StateType } from '../state/state.types';
import { toggleDrawer } from '../state/actions/scigateway.actions';
import { PluginConfig } from '../state/scigateway.types';
import { ListItemText } from '@material-ui/core';
import { MemoryRouter } from 'react-router';

describe('Navigation drawer component', () => {
  let shallow;
  let mount;
  let mockStore;
  let state: StateType;
  const routerState: RouterState = {
    action: 'POP',
    location: {
      hash: '',
      key: '',
      pathname: '/',
      search: '',
      state: {},
    },
  };

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'NavigationDrawer' });
    mount = createMount();
    mockStore = configureStore();
    state = {
      scigateway: initialState,
      router: routerState,
    };
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('Navigation drawer renders correctly when open', () => {
    state.scigateway.drawerOpen = true;

    const wrapper = shallow(<NavigationDrawer store={mockStore(state)} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Navigation drawer renders correctly when closed', () => {
    state.scigateway.drawerOpen = false;

    const wrapper = shallow(<NavigationDrawer store={mockStore(state)} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sends toggleDrawer action when chevron clicked', () => {
    const testStore = mockStore(state);
    const wrapper = mount(<NavigationDrawer store={testStore} />);

    wrapper.find('button').first().simulate('click');

    expect(testStore.getActions().length).toEqual(1);
    expect(testStore.getActions()[0]).toEqual(toggleDrawer());
  });

  function buildPlugin(
    order: number,
    displayName: string,
    section: string
  ): PluginConfig {
    return {
      order: order,
      displayName: displayName,
      section: section,
      link: 'link',
      plugin: 'plugin',
    };
  }

  it('renders a plugin list grouped by sections ordered alphabetically when open', () => {
    const dummyPlugins: PluginConfig[] = [
      buildPlugin(1, 'data-plugin', 'DATA'),
      buildPlugin(2, 'data-plugin2', 'DATA'),
      buildPlugin(1, 'analysis-plugin2', 'ANALYSIS'),
      buildPlugin(2, 'data-plugin1', 'DATA'),
      buildPlugin(-1, 'analysis-plugin', 'ANALYSIS'),
      {
        order: 3,
        plugin: 'data-plugin-no-displayname',
        link: 'plugin_link',
        section: 'DATA',
      },
    ];
    state.scigateway.plugins = dummyPlugins;
    state.scigateway.drawerOpen = true;

    const wrapper = shallow(<NavigationDrawer store={mockStore(state)} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('[to="plugin_link"]').first()).toMatchSnapshot();
  });

  it('renders a plugin with displayName but no logo or altText', () => {
    const dummyPlugins: PluginConfig[] = [
      {
        order: 0,
        plugin: 'data-plugin',
        link: 'plugin_link',
        section: 'DATA',
        displayName: '\xa0display name',
      },
    ];
    state.scigateway.plugins = dummyPlugins;
    state.scigateway.drawerOpen = true;

    const testStore = mockStore(state);
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <NavigationDrawer store={testStore} />
      </MemoryRouter>
    );

    expect(wrapper.find('img')).toHaveLength(0);
    const listItemText = wrapper.find(ListItemText).last();
    expect(listItemText.text()).toEqual('\xa0display name');
    expect(listItemText.prop('inset')).toBeTruthy();
  });

  it('renders a plugin with displayName and altText but no logo', () => {
    const dummyPlugins: PluginConfig[] = [
      {
        order: 0,
        plugin: 'data-plugin',
        link: 'plugin_link',
        section: 'DATA',
        displayName: '\xa0display name',
        logoAltText: 'DataGateway',
      },
    ];
    state.scigateway.plugins = dummyPlugins;
    state.scigateway.drawerOpen = true;

    const testStore = mockStore(state);
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <NavigationDrawer store={testStore} />
      </MemoryRouter>
    );

    expect(wrapper.find('img')).toHaveLength(0);
    const listItemText = wrapper.find(ListItemText).last();
    expect(listItemText.text()).toEqual('DataGateway\xa0display name');
    expect(listItemText.prop('inset')).toBeTruthy();
  });

  it('renders a plugin with logoLightMode', () => {
    const dummyPlugins: PluginConfig[] = [
      {
        order: 0,
        plugin: 'data-plugin',
        link: 'plugin_link',
        section: 'DATA',
        displayName: '\xa0display name',
        logoAltText: 'DataGateway',
        logoDarkMode: '/darkLogo.svg',
        logoLightMode: '/lightLogo.svg',
      },
    ];
    state.scigateway.plugins = dummyPlugins;
    state.scigateway.drawerOpen = true;

    const testStore = mockStore(state);
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <NavigationDrawer store={testStore} />
      </MemoryRouter>
    );

    expect(wrapper.find('img').prop('src')).toEqual('/lightLogo.svg');
    const listItemText = wrapper.find(ListItemText).last();
    expect(listItemText.text()).toEqual('\xa0display name');
    expect(listItemText.prop('inset')).toBeFalsy();
  });

  it('renders a plugin with logoDarkMode', () => {
    const dummyPlugins: PluginConfig[] = [
      {
        order: 0,
        plugin: 'data-plugin',
        link: 'plugin_link',
        section: 'DATA',
        displayName: '\xa0display name',
        logoAltText: 'DataGateway',
        logoDarkMode: '/darkLogo.svg',
        logoLightMode: '/lightLogo.svg',
      },
    ];
    state.scigateway.plugins = dummyPlugins;
    state.scigateway.drawerOpen = true;
    state.scigateway.darkMode = true;

    const testStore = mockStore(state);
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <NavigationDrawer store={testStore} />
      </MemoryRouter>
    );

    expect(wrapper.find('img').prop('src')).toEqual('/darkLogo.svg');
    const listItemText = wrapper.find(ListItemText).last();
    expect(listItemText.text()).toEqual('\xa0display name');
    expect(listItemText.prop('inset')).toBeFalsy();
  });
});
