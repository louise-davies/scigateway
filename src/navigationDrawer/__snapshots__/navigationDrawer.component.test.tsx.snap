// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation drawer component Navigation drawer renders correctly when closed 1`] = `
<ForwardRef(Drawer)
  PaperProps={
    Object {
      "sx": [Function],
    }
  }
  anchor="left"
  open={false}
  sx={
    Object {
      "flexShrink": 0,
      "width": [Function],
    }
  }
  variant="persistent"
>
  <ForwardRef(Box)
    boxSizing="border-box"
    display="flex"
    flexDirection="column"
    height="100%"
    justifyContent="flex-start"
  >
    <ForwardRef(Box)
      marginTop="auto"
    >
      <Styled(img)
        alt="alternative-text"
        src="stfc-logo-blue-text.png"
      />
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;

exports[`Navigation drawer component Navigation drawer renders correctly when open 1`] = `
<ForwardRef(Drawer)
  PaperProps={
    Object {
      "sx": [Function],
    }
  }
  anchor="left"
  open={true}
  sx={
    Object {
      "flexShrink": 0,
      "width": [Function],
    }
  }
  variant="persistent"
>
  <ForwardRef(Box)
    boxSizing="border-box"
    display="flex"
    flexDirection="column"
    height="100%"
    justifyContent="flex-start"
  >
    <ForwardRef(Box)
      marginTop="auto"
    >
      <Styled(img)
        alt="alternative-text"
        src="stfc-logo-blue-text.png"
      />
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;

exports[`Navigation drawer component does not display link to homepage if a homepage link is set 1`] = `
<ForwardRef(Drawer)
  PaperProps={
    Object {
      "sx": [Function],
    }
  }
  anchor="left"
  open={true}
  sx={
    Object {
      "flexShrink": 0,
      "width": [Function],
    }
  }
  variant="persistent"
>
  <ForwardRef(Box)
    boxSizing="border-box"
    display="flex"
    flexDirection="column"
    height="100%"
    justifyContent="flex-start"
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "color": [Function],
          "paddingBottom": 0,
          "paddingLeft": 2,
          "paddingTop": 2,
          "textAlign": "left",
        }
      }
      variant="h6"
    >
      DATA
    </ForwardRef(Typography)>
    <ForwardRef(List)
      component="nav"
    >
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-plugin_link"
        key="0"
        to="plugin_link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="display name"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
    </ForwardRef(List)>
    <ForwardRef(Box)
      marginTop="auto"
    >
      <Styled(img)
        alt="alternative-text"
        src="stfc-logo-blue-text.png"
      />
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;

exports[`Navigation drawer component does not render admin plugins or plugins that ask to hide in list 1`] = `
<ForwardRef(Drawer)
  PaperProps={
    Object {
      "sx": [Function],
    }
  }
  anchor="left"
  open={true}
  sx={
    Object {
      "flexShrink": 0,
      "width": [Function],
    }
  }
  variant="persistent"
>
  <ForwardRef(Box)
    boxSizing="border-box"
    display="flex"
    flexDirection="column"
    height="100%"
    justifyContent="flex-start"
  >
    <ForwardRef(Box)
      marginTop="auto"
    >
      <Styled(img)
        alt="alternative-text"
        src="stfc-logo-blue-text.png"
      />
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;

exports[`Navigation drawer component renders a plugin list grouped by sections ordered alphabetically when open 1`] = `
<ForwardRef(Drawer)
  PaperProps={
    Object {
      "sx": [Function],
    }
  }
  anchor="left"
  open={true}
  sx={
    Object {
      "flexShrink": 0,
      "width": [Function],
    }
  }
  variant="persistent"
>
  <ForwardRef(Box)
    boxSizing="border-box"
    display="flex"
    flexDirection="column"
    height="100%"
    justifyContent="flex-start"
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "color": [Function],
          "paddingBottom": 0,
          "paddingLeft": 2,
          "paddingTop": 2,
          "textAlign": "left",
        }
      }
      variant="h6"
    >
      ANALYSIS
    </ForwardRef(Typography)>
    <ForwardRef(List)
      component="nav"
    >
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-link"
        key="0"
        to="link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="analysis-plugin"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-link"
        key="1"
        to="link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="analysis-plugin2"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
    </ForwardRef(List)>
    <ForwardRef(Typography)
      sx={
        Object {
          "color": [Function],
          "paddingBottom": 0,
          "paddingLeft": 2,
          "paddingTop": 2,
          "textAlign": "left",
        }
      }
      variant="h6"
    >
      DATA
    </ForwardRef(Typography)>
    <ForwardRef(List)
      component="nav"
    >
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-link"
        key="0"
        to="link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="data-plugin"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-link"
        key="1"
        to="link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="data-plugin1"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-link"
        key="2"
        to="link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="data-plugin2"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
      <ForwardRef(ListItem)
        button={true}
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        dense={true}
        id="plugin-link-plugin_link"
        key="3"
        to="plugin_link"
      >
        <ForwardRef(ListItemText)
          inset={true}
          primary="data-plugin-no-displayname"
          primaryTypographyProps={
            Object {
              "sx": Object {
                "color": [Function],
                "fontWeight": "bold",
                "textAlign": "left",
              },
              "variant": "subtitle1",
            }
          }
        />
      </ForwardRef(ListItem)>
    </ForwardRef(List)>
    <ForwardRef(Box)
      marginTop="auto"
    >
      <Styled(img)
        alt="alternative-text"
        src="stfc-logo-blue-text.png"
      />
    </ForwardRef(Box)>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;

exports[`Navigation drawer component renders a plugin list grouped by sections ordered alphabetically when open 2`] = `
<ForwardRef(ListItem)
  button={true}
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  dense={true}
  id="plugin-link-plugin_link"
  key="3"
  to="plugin_link"
>
  <ForwardRef(ListItemText)
    inset={true}
    primary="data-plugin-no-displayname"
    primaryTypographyProps={
      Object {
        "sx": Object {
          "color": [Function],
          "fontWeight": "bold",
          "textAlign": "left",
        },
        "variant": "subtitle1",
      }
    }
  />
</ForwardRef(ListItem)>
`;
